src/java.base/share/classes/jdk/internal/util/jar/JarIndex.java

```diff
@@ -151,8 +151,8 @@ private void addToList(String key, String value,
      *
      * @param fileName the key of the mapping
      */
-    public LinkedList<String> get(String fileName) {
-        LinkedList<String> jarFiles = null;
+    public List<String> get(String fileName) {
```

Comment:
IcedTea-Web seems to be using this method reflectively:
https://github.com/AdoptOpenJDK/IcedTea-Web/blob/master/common/src/main/java/net/adoptopenjdk/icedteaweb/jdk89access/JarIndexAccess.java#L80

Reply:
Just for completeness, they're using the add-exports:
https://github.com/AdoptOpenJDK/IcedTea-Web/blob/master/launchers/itw-modularjdk.args#L19

Reply:
Should we then revert the changes to `JarIndex`?

Reply:
But don't people access these internal code at their own risk, as jdk may change these code at any time without notice?

Reply:
True, I just wonder whether it's OK to change internals when we know for sure that this breaks 3rd party code

Reply:
I think so. There are always unexpected ways the jdk may break and third-party libraries would need a different workaround for a new java version. For instance, in Apache log4j, its library has a special guard against the broken implementation of Reflection getCallerClass during java 7. The libraries can just handle these in their version-specific components.

Especially given the fact that the code linked above already has version-specific handling (8 vs 9), so it won't hurt much for them to add another piece of logic to handle jdk 17+ in case this optimization is merged.